packages/validator-ajv8/src/validator.ts

```diff
@@ -202,17 +204,66 @@ export default class AJV8Validator<
    * @private
    */
   private transformRJSFValidationErrors(
-    errors: ErrorObject[] = []
+    errors: ErrorObject[] = [],
+    uiSchema?: UiSchema<T, S, F>
   ): RJSFValidationError[] {
     return errors.map((e: ErrorObject) => {
-      const { instancePath, keyword, message, params, schemaPath } = e;
+      const {
+        instancePath,
+        keyword,
+        params,
+        schemaPath,
+        parentSchema,
+        ...rest
+      } = e;
+      let { message } = rest;
       let property = instancePath.replace(/\//g, ".");
-      let stack = `${property} ${message}`.trim();
+
+      let stack: string | undefined = undefined;
+
       if ("missingProperty" in params) {
         property = property
           ? `${property}.${params.missingProperty}`
           : params.missingProperty;
+
+        const currentProperty: string = params.missingProperty;
+
+        const uiSchemaTitle = get(
+          uiSchema,
+          `${property.replace(/^\./, "")}.['ui:title']`
+        );
+
+        if (uiSchemaTitle) {
+          message = message?.replace(currentProperty, uiSchemaTitle);
+        } else {
+          const parentSchemaTitle =
+            parentSchema?.[PROPERTIES_KEY]?.[currentProperty]?.title;
+
+          if (parentSchemaTitle) {
+            message = message?.replace(currentProperty, parentSchemaTitle);
+          }
+        }
+
         stack = message!;
+      } else {
+        let title = property.split(".").pop() as string;
+
+        const uiSchemaTitle = get(
+          uiSchema,
+          `${property.replace(/^\./, "")}.['ui:title']`
+        );
```

Comment:
```suggestion
        const uiSchemaTitle = getUiOptions<T, S, F>(get(uiSchema,`${property.replace(/^\./, "")}`)).title;
```

Reply:
Also, since you are doing this in both branches of the `if` statement, consider moving the variable assignment above and out of the `if`

Reply:
@heath-freenome the property paths are different in each case so the best way to generalise it would be to pull it into a function, I'm not sure that makes it better but I would do that if you agree.

Also, you don't need to pass the `<T, S, F>` type arguments when calling `getUiOptions`, typescript will infer those.

Reply:
Ah, yes, you are right in that property is updated in the first `if` block and isn't in the second.